IdentityCore/src/MSIDConstants.h

```diff
@@ -43,6 +43,30 @@ typedef NS_ENUM(NSInteger, MSIDWebviewType)
     MSIDWebviewTypeWKWebView,
 };
 
+typedef NS_ENUM(NSInteger, MSIDInteractiveRequestType)
+{
+    MSIDInteractiveRequestBrokeredType = 0,
+    MSIDInteractiveRequestLocalType
+};
+
+typedef NS_ENUM(NSInteger, MSIDUIBehaviorType)
+{
+    MSIDUIBehaviorInteractiveType = 0,
+    MSIDUIBehaviorAutoType,
+    MSIDUIBehaviorForceType
+};
+
+typedef NS_ENUM(NSInteger, MSIDPromptType)
+{
+    MSIDPromptTypePromptIfNecessary = 0, // No prompt specified, will use cookies is present, prompt otherwise
+    MSIDPromptTypeLogin, // prompt == "login", will force user to enter credentials
+    MSIDPromptTypeConsent, // prompt == "consent", will force user to grant permissions
+    MSIDPromptTypeSelectAccount, // prompt == "select_account", will show an account picker UI
+    MSIDPromptTypeRefreshSession, // prompt=refresh_session
+    MSIDPromptTypeNever, // prompt=none, ensures user is never prompted
+    MSIDPromptTypeDefault = MSIDPromptTypePromptIfNecessary
```

Comment:
just curious, why do we need MSIDPromptTypeDefault? if we don't provide value to enum, it will be MSIDPromptTypePromptIfNecessary anyway...

Reply:
If we use default in our code, it will be easier to change the default value in the future. If we use MSIDPromptTypePromptIfNecessary, we'd need to switch it in all APIs that we use.

Reply:
I see. If we change it, we should also make sure that default value is always first.